pytorch/training/docker/1.7.0/py3/Dockerfile.cpu

```diff
@@ -0,0 +1,122 @@
+ENV DGLBACKEND=pytorch
+
+ARG PT_TRAINING_URL=https://aws-pytorch-binaries.s3-us-west-2.amazonaws.com/r1.7.0_aws/20201023-100208/7626c0c94565c7f94d05dd25dda3b29500037017/cpu/torch-1.7.0-cp36-cp36m-manylinux1_x86_64.whl
+WORKDIR /
+
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends \
+    ca-certificates \
+    cmake \
+    curl \
+    emacs \
+    git \
+    wget \
+    vim \
+    build-essential \
+    jq \
+    libglib2.0-0 \
+    libgl1-mesa-glx \
+    libsm6 \
+    libxext6 \
+    libxrender-dev \
+    zlib1g-dev \
+    software-properties-common \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN wget https://www.open-mpi.org/software/ompi/v4.0/downloads/openmpi-$OPEN_MPI_VERSION.tar.gz \
+ && gunzip -c openmpi-$OPEN_MPI_VERSION.tar.gz | tar xf - \
+ && cd openmpi-$OPEN_MPI_VERSION \
+ && ./configure --prefix=/home/.openmpi \
+ && make all install \
+ && cd .. \
+ && rm openmpi-$OPEN_MPI_VERSION.tar.gz \
+ && rm -rf openmpi-$OPEN_MPI_VERSION
+
+# The ENV variables declared below are changed in the previous section
+# Grouping these ENV variables in the first section causes
+# ompi_info to fail. This is only observed in CPU containers
+ENV PATH="$PATH:/home/.openmpi/bin"
+ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/home/.openmpi/lib/"
+RUN ompi_info --parsable --all | grep mpi_built_with_cuda_support:value
+
+# Copy workaround script for incorrect hostname
+COPY changehostname.c /
+COPY start_with_right_hostname.sh /usr/local/bin/start_with_right_hostname.sh
+
+RUN curl -L -o ~/miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+ && chmod +x ~/miniconda.sh \
+ && ~/miniconda.sh -b -p /opt/conda \
+ && rm ~/miniconda.sh \
+ && /opt/conda/bin/conda install -y -c anaconda \
+    python=$PYTHON_VERSION \
+    numpy \
+    ipython \
+    mkl \
+    mkl-include \
+    cython \
+    typing \
+    "pyopenssl>=17.5.0" \
+ && conda install -c conda-forge \
+    opencv \
+ && conda install -y \
+    scikit-learn \
+    pandas \
+    h5py \
+    requests \
+ && conda install -c dglteam -y dgl \
+ && /opt/conda/bin/conda clean -ya \
+ && conda clean -ya
+
+# The following section uninstalls torch and torchvision before installing the
+# custom versions from an S3 bucket. This will need to be removed in the future
+RUN pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org \
+ && ln -s /opt/conda/bin/pip /usr/local/bin/pip3 \
+ && pip install --no-cache-dir -U \
+    "awscli<2" \
+    fastai==1.0.59 \
+    scipy==1.2.2 \
+    click \
+    smdebug==${SMDEBUG_VERSION} \
+    "sagemaker<2" \
+    sagemaker-experiments==0.* \
+    "sagemaker-pytorch-training<3" \
+    psutil==5.6.7 \
+    Pillow==7.1.0 \
+ && pip install --no-cache-dir -U ${PT_TRAINING_URL} \
+ && pip uninstall -y torchvision \
+ && pip install --no-deps --no-cache-dir -U \
+    https://choidong-dev.s3-us-west-2.amazonaws.com/torchvision-0.8.0/cpu/torchvision-0.8.0-cp36-cp36m-linux_x86_64.whl
```

Comment:
@ydaiming , we should create a new (temporary) torchvision binary, since we're working with CUDA 11 now.